src/main/java/org/assertj/core/api/InstanceOfAssertFactories.java

```diff
@@ -0,0 +1,70 @@
+package org.assertj.core.api;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.DoublePredicate;
+import java.util.function.IntPredicate;
+import java.util.function.LongPredicate;
+import java.util.function.Predicate;
+
+/**
+ * Static {@link InstanceOfAssertFactory InstanceOfAssertFactories} for {@link Assert#instanceOf(InstanceOfAssertFactory)}.
+ *
+ * @since 3.13.0
+ */
+public final class InstanceOfAssertFactories {
+
+  public static final InstanceOfAssertFactory<Predicate, PredicateAssert<Object>> PREDICATE = new InstanceOfAssertFactory<>(Predicate.class,
+                                                                                                                            PredicateAssert::new);
+
+  @SuppressWarnings("unchecked")
+  public static <T> InstanceOfAssertFactory<Predicate, PredicateAssert<T>> predicate(Class<T> type) {
+    return new InstanceOfAssertFactory<>(Predicate.class, predicate -> new PredicateAssert<>(predicate));
```

Comment:
use `PredicateAssert::new`

Reply:
same comment whenever we can use method reference

Reply:
For some reason the compiler was not able to infer the right return type in case of method reference, this is why I kept the lambda. I'll play a bit more with this, if I don't find a solution I'll put a suppress warning and a comment on these cases.